test/karma.config.mjs

```diff
@@ -0,0 +1,167 @@
+import chalk from 'chalk';
+import { startGoldensServer } from './GoldensServer.mjs';
+
+// import { babel } from '@rollup/plugin-babel';
+// import json from '@rollup/plugin-json';
+// import ts from '@rollup/plugin-typescript';
+// import { nodeResolve } from '@rollup/plugin-node-resolve';
+// import commonjs from '@rollup/plugin-commonjs';
+//
+// const plugins = [
+//   json(),
+//   ts(),
+//   babel({
+//     extensions: ['.ts', '.js'],
+//     babelHelpers: 'bundled',
+//   }),
+//   nodeResolve(),
+//   commonjs({ include: 'node_modules/**' })
+// ];
+
+// https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables
+const CI = !!process.env.CI
+const BROWSERS = process.env.BROWSERS ? process.env.BROWSERS.split(/,| /g) : ['chrome', 'firefox'];
+const TEST_FILES = process.env.TEST_FILES ? process.env.TEST_FILES.split(/,| /g) : null;
+const VISUAL_TEST_CONFIG = {
+  recreate: Number(process.env.QUNIT_RECREATE_VISUAL_REFS),
+  debug: Number(process.env.QUNIT_DEBUG_VISUAL_TESTS),
+};
+
+/** 
+ * http://karma-runner.github.io/6.4/config/configuration-file.html
+ * https://github.com/tom-sherman/blog/blob/main/posts/02-running-jest-tests-in-a-browser.md
+ * @param {*} config 
+ */
+export default async function (config) {
+  const browsers = (CI ?
+    ['ChromeCI', 'FirefoxCI'] :
+    [...(!VISUAL_TEST_CONFIG.debug && !VISUAL_TEST_CONFIG.recreate ? ['ChromeCI', 'FirefoxCI'] : []), 'Chrome', 'Firefox'])
+    .filter(browser => BROWSERS.some(b => browser.toLowerCase().startsWith(b)));
+  if (VISUAL_TEST_CONFIG.debug || VISUAL_TEST_CONFIG.recreate) {
+    if (browsers.length > 1) {
+      console.warn(chalk.yellow(`Debugging/recreating visual tests is allowed ONLY when running tests in a single browser`));
+    }
+    if (CI) {
+      throw new Error(chalk.red(`Debugging/recreating visual refs is banned in CI`));
+    }
+  }
+  if (browsers.length === 0) {
+    throw new Error(`no browsers to launch, input: ${BROWSERS}`);
+  }
+  
+  config.set({
+    plugins: [
+      'karma-jasmine',
+      'karma-qunit',
+      'karma-chrome-launcher',
+      'karma-firefox-launcher',
+      'karma-coverage',
+      'karma-spec-reporter'
+    ],
+
+    // base path that will be used to resolve all patterns (eg. files, exclude)
+    basePath: '../',
+    autoWatch: false,
+    singleRun: true,
+    pingTimeout: CI ? 10000 : 2000,
+    browserDisconnectTimeout: CI ? 10000 : 2000,
+
+    // frameworks to use
+    // available frameworks: https://npmjs.org/browse/keyword/karma-adapter
+    frameworks: [/*'jasmine',*/ 'qunit'],
+    browsers,
+    customLaunchers: {
+      ChromeCI: {
+        base: 'Chrome',
+        flags: [
+          '--no-sandbox',
+          '--disable-gpu',
+          '--disable-translate',
+          '--disable-extensions',
+          '--remote-debugging-port=9222',
+        ],
+      },
+      FirefoxCI: {
+        base: 'Firefox'
+      }
```

Comment:
I tested with base `ChromeHeadless` and `FirefoxHeadless`
It seems there is no difference as long as CI uses `xvfb-run` but I might be wrong